Add back examples updated for mbed-os-5.10

adbridge opened the pull request:
### Description

While feature changes were being added to mbed-os a number of
examples were removed from this list due to incompatibility issues.
This PR adds those examples back in:
Filesystem,
Blockdevice,
Mesh-minimal,
Bootloader.

This PR also adds in the new NFC example.

### Pull request type

    [x] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Breaking change

adbridge committed:
Add back examples updated for mbed-os-5.10

While feature changes were being added to mbed-os a number of
examples were removed from this list due to incompatibility issues.
This PR adds those examples back in:
Filesystem,
Blockdevice,
Mesh-minimal,
Bootloader.

This PR also adds in the new NFC example.

cmonr commented:
@adbridge Two things:
1) A lot of the examples are only built against the K64F, leading to longer build times for just that target. Do we want to consider targeting something different so that the risk of having K64F builds timing out remains low?
2) If a target is not specified, the example is built against all available targets (in reference to this: https://github.com/ARMmbed/mbed-os/pull/8246/files#diff-93b92c61c9977ce2503bf0c37f41a92dR303). Is that what we want for the NFC example?

cmonr commented:
@ARMmbed/mbed-os-maintainers ^^^

adbridge commented:
@cmonr In terms of 1. That is what is specified by the example owners so we shouldn't change that unless they say so. In terms of 2. I will ask Vincent to review.
@pan- @donatieng  Can you please comment on the compilation for the NFC example.

0xc0170 reviewed (approved):
This is the revert of the removal earlier - LGTM

As @cmonr stated, we should review this for 5.11. @OPpuolitaival If not yet in, we should review example testing

pan- commented on tools/test/examples/examples.json:
```diff
@@ -229,6 +229,83 @@
+        "name": "mbed-os-example-nfc",
+        "github": "https://github.com/ARMmbed/mbed-os-example-nfc",
+        "mbed": [
+            "https://os.mbed.com/teams/mbed-os-examples/code/mbed-os-example-nfc-EEPROM/",
```
That example is restricted to `NUCLEO_F401RE` and the `DISCO_L475VG_IOT01A ` at the current minute.

0xc0170 replied:
> That example is restricted to NUCLEO_F401RE and the DISCO_L475VG_IOT01A at the current minute.

Not certain if this one commit just adds back what used to be there or adds something new. Should be clarified and fixed

cmonr replied:
@0xc0170 This PR is doing both. NFC is a new example that was introduced with 5.10.

OPpuolitaival commented on tools/test/examples/examples.json:
```diff
@@ -229,6 +229,83 @@
+        ],
+        "test-repo-source": "mbed",
+        "features" : [],
+        "targets" : [],
```
That really looks that it does not work for all devices as configured now

Commit:
Restricted nfc example to two targets

cmonr commented:
@pan- @donatieng Mind taking another look? I've restricted the NFC example to the two specified targets.

pan- reviewed (approved):
Approved for NFC.

cmonr commented:
/morph build

cmonr commented:
Going to retry. Not sure why the examples failed to build.
/morph build

cmonr commented:
The issue is subtle, but it looks like the build-prep step is failing to clone the nfc examples.

Commit:
Removed trailing slashes in NFC examples URL

This apparently breaks the examples.py script.

cmonr commented:
/morph build

cmonr commented:
For reference, it appears that having the trailing slash was causing the following error:
```
18:08:57 '' example directory doesn't exist. Skipping...
18:08:57 '' example directory doesn't exist. Skipping...
```

However, the script continues without failing.

studavekar commented:
> # [Build : FAILURE](https://mbed-os.mbedcloudtesting.com/job/mbed-os-build-matrix/3172/)
> Build number : 3172
> Build artifacts/logs : http://mbed-os.s3-website-eu-west-1.amazonaws.com/?prefix=builds/8246/

Looks like legit build failures 

1)  **atmel-rf' is not unique**
```
ConfigException: Library name 'atmel-rf' is not unique (defined in '/builds/ws/mbed-os-build-matrix/3172/K64F_GCC_ARM/sources/examples/mbed-os-example-tls-tls-client/easy-connect/atmel-rf-driver/mbed_lib.json' and '/builds/ws/mbed-os-build-matrix/3172/K64F_GCC_ARM/sources/examples/mbed-os-example-tls-tls-client/mbed-os/components/802.15.4_RF/atmel-rf-driver/mbed_lib.json')
```
2) **Output: ./main.cpp:41:1: error: 'SPIFBlockDevice' does not name a type**

```
[DEBUG] Output: ./main.cpp:41:1: error: 'SPIFBlockDevice' does not name a type
[DEBUG] Output:  SPIFBlockDevice bd(
[DEBUG] Output:  ^~~~~~~~~~~~~~~
[DEBUG] Output: ./main.cpp: In function 'void erase()':
[DEBUG] Output: ./main.cpp:56:15: error: 'bd' was not declared in this scope
[DEBUG] Output:      int err = bd.init();
[DEBUG] Output:                ^~
[DEBUG] Output: ./main.cpp: In function 'int main()':
[DEBUG] Output: ./main.cpp:91:25: error: 'bd' was not declared in this scope
[DEBUG] Output:      int err = fs.mount(&bd);
[DEBUG] Output:                          ^~
```

cmonr commented:
Pinging @ARMmbed/mbed-os-storage (and @geky) for the filesystem example.

@ARMmbed/mbed-os-maintainers Who do we ping for mbed-os-example-tls-tls-client?

0xc0170 commented:
> @ARMmbed/mbed-os-maintainers Who do we ping for mbed-os-example-tls-tls-client?

Can you review the failures above @k-stachowiak @Patater 

> Pinging @ARMmbed/mbed-os-storage (and @geky) for the filesystem example.

I suspect most of the storage team is OoO today, @geky can you help in case we do not receive a response here (I contacted them privately as well)

k-stachowiak commented:
@0xc0170 In mbed-os-example-tls we need this merged: https://github.com/ARMmbed/mbed-os-example-tls/pull/198 It is pending minor changes from @SeppoTakalo. Please let me know if we may do something to support him.

yossi2le commented:
@0xc0170 The fileststem example should be compiled on K82F after the recent changes.

k-stachowiak commented:
https://github.com/ARMmbed/mbed-os-example-tls/pull/198 has just been merged.

cmonr commented:
Rerunning to verify TLS example was fixed, as well as to isolate Filesystem errors.

/morph build

geky commented:
Hmm, @yossi2le, the examples likely used with storage not normally on the devices. Should we add "components_add" to the mbed_app.json to make all storage components available in the storage examples?

geky commented:
Proposed fix for filesystems: https://github.com/ARMmbed/mbed-os-example-filesystem/pull/69, https://github.com/ARMmbed/mbed-os-example-blockdevice/pull/14, unfortunately I can't test it until later today.

Also +1 to @yossi2le's point that testing storage on K82F or nRF52-DK is more useful than K64F.

geky commented:
From discussion on https://github.com/ARMmbed/mbed-os-example-filesystem/pull/69, those prs do not fix this issue. The fix is using MX25R or nRF52-DK instead of K64F. The K64F simply does not have SPIF on board.

cmonr commented:
@geky @deepikabhavnani Looks like the NRF52_DK wouldn't pass even if the target was changed. See attached log.

Also, the MX25R? Is that a target??

[log.txt](https://github.com/ARMmbed/mbed-os/files/2424997/log.txt)

yossi2le commented:
In the past it was running on K64F only. Can we change it to K82F? It got SPIF and it has all the properties already done in targets.json.

geky commented:
Sorry, that was a typo on my part. +1 for K82F

Commit:
Replace storage example targets from K64F to K82F

Suggested here: https://github.com/ARMmbed/mbed-os/pull/8246#issuecomment-425199620

cmonr commented:
Fingers crossed this does  the trick. Otherwise, I'm going to keep those two storage examples disabled and bring this in without them to get 5.10.1 generated tonight.

/morph build

(I say fingers crossed because even after changing targets, the build still failed locally, similar to this failure: https://github.com/ARMmbed/mbed-os/pull/8246#issuecomment-425162349)

deepikabhavnani commented:
K64F": {
        "supported_form_factors": ["ARDUINO"],
        "components": ["SD"],

Similar to this all storage components should be present in K82F else will get build failures

Commit:
Remove block device example in the short term

cmonr commented:
Well darn. Looks like the block device example still needs some work.

Going to progress this without it for now.

/morph build

NirSonnenschein commented:
/morph export-build

NirSonnenschein commented:
This PR is not passing export due to failures finding printf on the following boards (failure occurs on all toolchains but only occurs on two boards ). 
1. mbed-os-example-nfc-EEPROM DISCO_L475VG_IOT01A on  DISCO_L475VG_IOT01A 
2. mbed-os-example-nfc-EEPROM NUCLEO_F401RE on NUCLEO_F401RE
@donatieng and @pan- @adbridge  has asked that you please take a look. Note: this is the LAST remaining PR for 5.10.1 which is supposed to be built later today.

also CCing @cmonr in case this runs into your shift.

Commit:
Removed mbed-os-example-nfc-EEPROM from examples.json

Last thing blocking 5.10.1 release. Example will need to be fixed before it can be brought in.

cmonr commented:
/morph build

cmonr commented:
/morph test
